src/d_main.c

```diff
@@ -255,7 +261,7 @@ void D_Display (void)
     case GS_LEVEL:
       if (!gametic)
         break;
-      if (automapactive)
+      if (automapactive && !automapoverlay)
```

Comment:
For your future sanity, I recommend abstracting this pair of conditions while you're introducing it, as it ends up being a common check. It's like this in dsda-doom:
```c
// doomstat.h
#define automap_on (automap_active && !automap_overlay)
#define automap_off (!automap_active || automap_overlay)
```
Not the most precise name but this was an existing variable at the time so I kept it.
It keeps the code closer to its original intent and readability at least, to me.

Reply:
Maybe we could replace the names with something along the lines of `automap_solid` and `automap_transparent`? Personally, I like the former but not the latter, it seems pretty long. If we _are_ going to add these, I'd like to settle which names should be used.

Reply:
Yes, I like the idea of abstracting the `automapactive` and `automapoverlay` checks into a pair of macros. But, since this would touch a lot of unrelated code, I'd also prefer this in a separate PR.